docs/api.md

```diff
@@ -0,0 +1,128 @@
+# API
+
+## Config Management API
+
+Grafana Cloud Agent exposes a REST API for managing instance configurations when
+it is running in scraping service mode. The following endpoints are exposed:
+
+- List configs: [`GET /agent/api/v1/configs`](#list-configs)
+- Get config: [`GET /agent/api/v1/configs/{name}`](#get-config)
+- Update config: [`PUT /agent/api/v1/config/{name}`](#update-config)
+- Delete config: [`DELETE /agent/api/v1/config/{name}`](#delete-config)
+
+### API Response
+
+All Config Management API endpoints will return responses in the following
+form, unless an internal service error prevents the server from responding
+properly:
+
+```json
+{
+  "status": "success" | "error",
+  "data": {}
+}
+```
+
+Status will be either `success` or `error`. All 2xx responses will be
+accompanied with a `success` value for the status field. 4xx and 5xx
+responses will provide a value of `error`.
+
+The data field may or may not be present, depending on the endpoint. It
+provides extra information for the query. The documentation for each endpoint
+will describe the full response provided.
+
+### List Configs
+
+```
+GET /agent/api/v1/configs
+```
+
+List Configs returns a list of the named configurations currently known by the
+underlying KV store.
+
+Response:
+
+```
+{
+  "status": "success",
+  "data": {
+    "configs": [
+      // list of config names:
+      "a",
+      "b",
+      "c",
+      // ...
+    ]
+  }
+}
+```
+
+### Get Config:
+
+```
+GET /agent/api/v1/configs/{name}
+```
+
+Get Config will return a single configuration by name. The configuration must
+exist or an error will be returned.
+
+Response on success:
+
+```
+{
+  "status": "success",
+  "data": {
+    "value": "/* configuration in the string */"
+  }
+}
+```
+
+By default, the configuration contained in "value" will be YAML, even if it was
+initially stored as JSON. To get back a JSON config string, add a
+`Content-Type: application/json` header to the request.
```

Comment:
I'm not sure how I feel about using the content type as an indication for part of the response. FWIW the response is still JSON with a single field being a YAML string. I'm going to file this one under nits as I don't think there's anything too egregious but keep it mind.